### PR TITLE
Add the ability to edit fields within a config map.

### PR DESCRIPTION
**Pull request by brendandburns**
Addresses part of https://github.com/kubernetes/kubernetes/issues/36222

Example command:

```console
$ kubectl edit configmap foo --config-map-data=bar
```

Will open the data element named `bar` in the `ConfigMap` named `foo` in `$EDITOR`, the edited contents are then updated back to the config map.

@kubernetes/sig-cli 

```release-note
Add a special purpose tool for editing individual fields in a ConfigMap with kubectl
```

### CODE REVIEW

**Comment by fabianofranz** [comment]
@brendandburns does `edit configmap` being a subcommand breaks the RESOURCE/NAME syntax (with slash)? Like in `$ kubectl edit configmap/foo`.

**Comment by brendandburns** [comment]
@fabianofranz it won't break it, but it will make this particular command not work...  

Subcommand seemed cleaner than a type-specific hack in `edit.go` but I'm open to other ideas.

**Comment by juanvallejo** [comment]
@brendandburns would `pkg/kubectl/cmd/edit_configmap.go` benefit from having a `RunEditConfigmap` function? `RunEdit` could also be called from there, and the command's `Run: func(...` might instead look something like:

```
Run: func(cmd *cobra.Command, args []string) {
   cmdutil.CheckErr(RunEditConfigmap(args))
}
```

This might make this file a bit easier to read

**Comment by brendandburns** [comment]
@juanvallejo done.

@fabianofranz any chance for another look at this?

Thanks!
--brendan

**Comment by brendandburns** [comment]
@k8s-bot gci gke e2e test this.

**Comment by brendandburns** [comment]
@k8s-bot cvm gke e2e test this

**Comment by brendandburns** [comment]
@k8s-bot unit test this

**Comment by brendandburns** [comment]
@k8s-bot unit test this

**Comment by fabianofranz** [comment]
@k8s-bot unit test this

**Comment by brendandburns** [comment]
@k8s-bot gci gke e2e test this

**Comment by brendandburns** [comment]
@k8s-bot gci gke e2e test this

**Comment by brendandburns** [comment]
@fabianofranz I think this is ready to go...

Thanks!

**Comment by fabianofranz** [comment]
@brendandburns Mind adding a release note? I think this is worth one, tks!

**Comment by brendandburns** [comment]
@fabianofranz release note added.

Thanks

**Comment by fabianofranz** [comment]
Tks!

File: pkg/kubectl/cmd/edit.go [hunk 1]
```diff
@@ -100,12 +100,20 @@ func NewCmdEdit(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
 		Long:    editLong,
 		Example: fmt.Sprintf(editExample),
 		Run: func(cmd *cobra.Command, args []string) {
+			args = append([]string{"configmap"}, args...)
```
Comment (liggitt) [inline pkg/kubectl/cmd/edit.go]: this breaks edit for all other resources

Comment (liggitt) [inline pkg/kubectl/cmd/edit.go]:
```
$ kubectl edit serviceaccount default
Error from server (NotFound): configmaps "serviceaccount" not found
```

**Comment by liggitt** [comment]
this broke edit for all resources other than configmaps, @fabianofranz can you investigate how this passed kubectl tests?

also, `edit` is a generic command, we can't add subcommands that claim the `configmap` space

reverting in https://github.com/kubernetes/kubernetes/pull/40421

**Comment by ymqytw** [comment]
Since this PR has been reverted in https://github.com/kubernetes/kubernetes/pull/40421.
/release-note-none

**Comment by KIVagant** [comment]
@brendandburns , any chance to get this feature back? It would be very useful.

**Comment by josefmeyer89** [comment]
Any change we can revisit this feature? It would be quite useful to have...

**Pull request merged**